recipes/volume_extend.rb

```diff
@@ -0,0 +1,63 @@
+#
+# Cookbook Name:: gluster
+# Recipe:: server_setup
+#
+# Copyright 2015, Andrew Repton
+# Copyright 2015, Biola University
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+node['gluster']['server']['volumes'].each do |volume_name, volume_values|
+  # 1. Get the current size of the logical volume
+  # 2. Compare to the size set for the gluster volume
+  # 3. If different, run a resize action against that volume
+  # ToDO: change hardcoded VG name gluster into an attribute
+  require 'lvm'
```

Comment:
> Shouldn't this be a node.default? And why not put it in the attributes file?

`node.set` is an alias for `node.default` so either way works, take your pick :)
You could put it in the attributes file, but most people are probably not going to look their for attributes set for other cookbooks other than the one they are looking at.  Never seen attributes for different cookbooks mixed together like that anyone.  If you have any example, please share

> Could you make the other changes a second PR?

Not if you want kitchen tests to pass :)